### PR TITLE
Update toolz to 0.8.2 and use tlz

### PR DESCRIPTION
**Pull request by groutr**
- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

This PR updates the minimum version of toolz for dask from 0.7.3 to 0.8.2.  It then replaces all imports for toolz with imports for tlz which prefers cytoolz if installed.

This also bumps cloudpickle from 0.2.1 to 0.2.2, the first version with support for Python 3.6 (which is the minimum supported version of Python by dask).

crossref: https://github.com/dask/dask/pull/5949#issuecomment-595766947

### CODE REVIEW

File: dask/tests/test_delayed.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
 from collections import namedtuple
 from operator import add, setitem
+from functools import partial
 import pickle
 from random import random
 import types
 
-from toolz import identity, partial, merge
+from tlz import merge
+from toolz import identity
```
Comment (TomAugspurger) [inline dask/tests/test_delayed.py]: Is identity available from tlz?

```suggestion
from tlz import merge, identity
```

Comment (groutr) [reply dask/tests/test_delayed.py]: It's used in the test `test_name_consistent_across_instances`.  The identity function from toolz generates a different key from the one in cytoolz. 

If any function could be used, perhaps use a pure function from the standard library?

File: dask/base.py [hunk 2]
```diff
@@ -717,13 +717,13 @@ def normalize_function(func):
 
 
 def _normalize_function(func):
-    if isinstance(func, curry):
+    if isinstance(func, curry) and hasattr(func, "_partial"):
```
Comment (jrbourbeau) [inline dask/base.py]: Why was this change needed?

File: dask/base.py [hunk 3]
```diff
@@ -717,13 +717,13 @@ def normalize_function(func):
...
         func = func._partial
     if isinstance(func, Compose):
         first = getattr(func, "first", None)
         funcs = reversed((first,) + func.funcs) if first else func.funcs
         return tuple(normalize_function(f) for f in funcs)
-    elif isinstance(func, partial):
+    elif isinstance(func, (partial, curry)):
```
Comment (jrbourbeau) [inline dask/base.py]: Same questions here, though it may be related to the other change made in this module

File: docs/source/install.rst [hunk 4]
```diff
@@ -86,39 +86,39 @@ These optional dependencies and their minimum supported versions are listed belo
 +-------------+----------+--------------------------------------------------------------+
 | Dependency  | Version  |                          Description                         |
 +=============+==========+==============================================================+
-|    bokeh    | >=1.0.0  |                Visualizing dask diagnostics                  |
+|     bokeh     | >=1.0.0  |                Visualizing dask diagnostics                  |
```
Comment (jrbourbeau) [inline docs/source/install.rst]: It looks like additional spaces were added which is causing some misalignment in the column width. Could you update to make sure each row has the same width for the "Dependency", "Version", and "Description" columns? Otherwise sphinx won't render the table:

Screen Shot 2020-03-10 at 12 19 18 PM

Comment (groutr) [reply docs/source/install.rst]: Hopefully, it's fixed now.

File: dask/base.py (hunk 2 above)
Comment (groutr) [reply dask/base.py]: Because cytoolz.curry and toolz.curry are not interchangeable. They have different attribute interfaces.

File: dask/base.py (hunk 3 above)
Comment (groutr) [reply dask/base.py]: See comment above.  I believe this is a bug in toolz, which I will be submitting a PR shortly to fix.

File: dask/base.py (hunk 2 above)
Comment (groutr) [reply dask/base.py]: Looks like I mis-spoke. Looking at it further with fresher eyes.

Comment (groutr) [reply dask/base.py]: Looks like we can remove that if statement entirely!

File: dask/base.py (hunk 3 above)
Comment (groutr) [reply dask/base.py]: We can check for curry here because curry and partial have the same `func`, `args`, and `keywords` attributes.

File: docs/source/install.rst (hunk 4 above)
Comment (jrbourbeau) [reply docs/source/install.rst]: Yep, looks good now. Thanks for updating
